packages/manager/src/features/CloudPulse/Alerts/CreateAlert/Criteria/DimensionFilterValue/useFirewallFetchOptions.ts

```diff
@@ -70,27 +83,56 @@ export function useFirewallFetchOptions(
 
   const idFilter = {
     '+or': filteredFirewallParentEntityIds.length
-      ? filteredFirewallParentEntityIds.map((id) => ({ id }))
+      ? filteredFirewallParentEntityIds.map((entity) => ({
+          id: entity.id,
+        }))
       : [{ id: '' }],
   };
 
-  const combinedFilter: Filter = {
+  const labelFilter = {
+    '+or': filteredFirewallParentEntityIds.length
+      ? filteredFirewallParentEntityIds.map((entity) => ({
+          label: entity.label,
+        }))
+      : [{ label: '' }],
+  };
+
+  const combinedFilterLinode: Filter = {
     '+and': [idFilter, regionFilter].filter(Boolean) as Filter[],
   };
 
+  const combinedFilterNodebalancer: Filter = {
+    '+and': [labelFilter, regionFilter].filter(Boolean) as Filter[],
+  };
+
   // Fetch all linodes with the combined filter
   const {
     data: linodes,
     isError: isLinodesError,
     isLoading: isLinodesLoading,
   } = useAllLinodesQuery(
     {},
-    combinedFilter,
+    combinedFilterLinode,
     filterLabels.includes(dimensionLabel ?? '') &&
-      filteredFirewallParentEntityIds.length > 0 &&
+      filteredFirewallParentEntityIds?.length > 0 &&
+      (!dashboardId || dashboardId === 4) &&
       supportedRegionIds?.length > 0
   );
 
+  // Fetch all nodebalancers with the combined filter
+  const {
+    data: nodebalancers,
+    isError: isNodebalancersError,
+    isLoading: isNodebalancersLoading,
+  } = useAllNodeBalancersQuery(
+    filterLabels.includes(dimensionLabel ?? '') &&
+      filteredFirewallParentEntityIds?.length > 0 &&
+      (!dashboardId || dashboardId === 8) &&
```

Comment:
```suggestion
      (!dashboardId || dashboardId === 8) &&
```

don't create id dependency in common functions, drive it through different parameters